components/hab/src/command/origin/delete.rs

```diff
@@ -27,8 +29,19 @@ pub fn start(ui: &mut UI, bldr_url: &str, token: &str, origin: &str) -> Result<(
 
     ui.status(Status::Deleting, format!("origin {}.", origin))?;
 
-    api_client.delete_origin(origin, token)
-              .map_err(Error::APIClient)?;
+    match api_client.delete_origin(origin, token) {
+        Ok(_) => (),
```

Comment:
Moving the happy path into the `Ok` match arm:
```suggestion
        Ok(_) => {
            ui.status(Status::Deleted, format!("origin {}.", origin))
              .map_err(Into::into)
        }
```
simplifies the logic and allows the elimination of the `return`s in the `Err` arms.